drivers/gpio/gpio_shakti.c

```diff
@@ -0,0 +1,355 @@
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <soc.h>
+#include <zephyr/drivers/gpio.h>
+#include <stdio.h>
+
+#define DT_DRV_COMPAT shakti_gpio
+
+#ifdef CONFIG_BOARD_SECURE_IOT
+
+#define CLOCK_FREQUENCY_SIM 1000000000
+#define CLOCK_FREQUENCY 40000000
+#endif
+
+
+#define GPIO_START 0x00040200 //GPIO Start Address */
+#define GPIO_OFFSET 0x08 /*!Generic offset used to access GPIO registers*/
+
+
+#define GPIO_DIRECTION_CNTRL_REG  (GPIO_START + (0 * GPIO_OFFSET ))
+#define GPIO_DATA_REG             (GPIO_START + (1 * GPIO_OFFSET ))
+#define GPIO_SET_REG              (GPIO_START + (2 * GPIO_OFFSET ))
+#define GPIO_CLEAR_REG            (GPIO_START + (3 * GPIO_OFFSET ))
+#define GPIO_TOGGLE_REG           (GPIO_START + (4 * GPIO_OFFSET ))
+#define GPIO_QUAL_REG             (GPIO_START + (5 * GPIO_OFFSET ))
+#define GPIO_INTERRUPT_CONFIG_REG (GPIO_START + (6 * GPIO_OFFSET ))
+
```

Comment:
No Space before close parenthesis ')'. must be (x * GPIO_OFFSET).

Reply:
Done